js/controller.js

```diff
@@ -630,17 +630,28 @@ function controller(OO, _, $) {
       if (videoElement) {
         videoElement.addEventListener('loadedmetadata', this.metaDataLoaded.bind(this));
 
-        // add the AirPlay TargetAvailability event listener
-        if (window.WebKitPlaybackTargetAvailabilityEvent && this.state.isAirplayAllowed) {
-          videoElement.addEventListener('webkitplaybacktargetavailabilitychanged',
-            _.bind(this.airPlayListener, this));
+        // add the AirPlay event listeners
+        if (window.WebKitPlaybackTargetAvailabilityEvent) {
```

Comment:
handling airplay functionality grows on so I suggest move it from `onVcVideoElementCreated` to a separate method